### DISCUSSION TITLE
Visualize the connection of php files.

### DISCUSSION THREAD

**Discussion by bzanios** [microsoft/vscode-discussions#874] [Extension Development QnA] [answered]
Hello my friends.
I would like to know if there is a way through VS Code (or some other way that you can propose) , to visualize the connections between the php files in a project. To explain better, i want a way to know the connections (visualize) with the other files and the connection that this files have with other files etc. For Example : The test1.php file requires the needed1.php and needed2.php files and these two files requiring a common file common1.php. How can i visualize my projetcs this way. Thank you for your responses.

**Comment by alefragnani** [comment, marked as answer]
Hi @bzanios ,

There is no such feature in VS Code, and taking a quick look in the marketplace, I didn't find any extension that does that either. If you know any npm package or python library that does that (generating an SVG for instance), you probably should be able create a new extension with it. It would be a great addition.

Outside VS Code, I remember two tools that I have used a lot, that you could take a look:
* Understand (https://scitools.com/#graphs)
* Structure101 (https://structure101.com/products/studio/)

Both tools does what you need, and a bit more, allowing you to understand your code architecture, collect metrics and so on. But right now, I don't remember if any of these tools allows you to export images of the connection graph.

Hope this helps

ps: By the way, for future feature requests about VS Code features and usage, please use the VS Code repository instead (https://github.com/microsoft/vscode/issues/new/choose). This Discussions repo is focused for extension development for now.
